src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -38,9 +38,34 @@ public QueryServiceConnection(String url, Properties properties) {
         this.properties = properties;
         this.serviceRootUrl = url.substring(Constants.DATASOURCE_TYPE.length());
         this.properties.put(Constants.LOGIN_URL, serviceRootUrl);
+        setupDefaultClientSecretsIfRequired(serviceRootUrl, this.properties);
         this.queryExecutor = new QueryExecutor(properties);
     }
 
+    private void setupDefaultClientSecretsIfRequired(String serviceRootUrl, Properties properties) {
+        if(properties.containsKey(Constants.USER) && !properties.containsKey(Constants.USER_NAME)) {
+            properties.put(Constants.USER_NAME, properties.get(Constants.USER));
+        }
+
+        if(properties.containsKey(Constants.USER_NAME)
+                && !properties.containsKey(Constants.CLIENT_ID)
```

Comment:
Can we add this in tokenHelper as these are related to tokens?

Reply:
Felt it is more appropriate in connection object that gives token helper the way/format it needs.